quadriga/src/main/java/edu/asu/diging/quadriga/web/service/impl/GraphCreationServiceImpl.java

```diff
@@ -0,0 +1,262 @@
+            	    		graphNodes, graphEdges, uniqueNodes, validEventGraph.getId().toString());
+            	}
+            	catch (ConceptpowerNoResponseException e) {
+            	    logger.error("Could not create graph for id : "+validEventGraph.getId(), e);
+            	}
+            });
+
+        GraphElements graphElements = new GraphElements();
+        graphElements.setNodes(wrapInGraphElements(graphNodes));
+        graphElements.setEdges(wrapInGraphElements(graphEdges));
+
+        return graphElements;
+    }
+
+    @Override
+    public String createNodesAndEdges(RelationEvent event, List<GraphData> graphNodes, List<GraphData> graphEdges,
+            Map<String, GraphNodeData> uniqueNodes, String eventGraphId) throws ConceptpowerNoResponseException {
+        Relation relation = event.getRelation();
+        String predicateNodeId = null;
+        String subjectNodeId = null;
+        String objectNodeId = null;
+
+        if (relation.getPredicate() != null) {
+            predicateNodeId = createPredicateNode(graphNodes, relation.getPredicate(), eventGraphId);
+        } else {
+            logger.error("A predicate is missing in one of the relations for EventGraph: " + eventGraphId);
+            return null;
+        }
+
+        if (relation.getSubject() != null) {
+            if (relation.getSubject() instanceof RelationEvent) {
+                subjectNodeId = createNodesAndEdges((RelationEvent) relation.getSubject(), graphNodes, graphEdges,
+                        uniqueNodes, eventGraphId);
+            } else {
+                subjectNodeId = createSubjectOrObjectNode(graphNodes, (AppellationEvent) relation.getSubject(),
+                        uniqueNodes, GraphNodeType.SUBJECT, eventGraphId);
+            }
+        } 
+
+
+        if (relation.getObject() != null) {
+            if (relation.getObject() instanceof RelationEvent) {
+                objectNodeId = createNodesAndEdges((RelationEvent) relation.getObject(), graphNodes, graphEdges,
+                        uniqueNodes, eventGraphId);
+            } else {
+                objectNodeId = createSubjectOrObjectNode(graphNodes, (AppellationEvent) relation.getObject(),
+                        uniqueNodes, GraphNodeType.OBJECT, eventGraphId);
+            }
+        } 
+
+
+        if(subjectNodeId != null && predicateNodeId != null) {
+            createEdge(graphEdges, subjectNodeId, predicateNodeId, eventGraphId);
+        }
+        
+        if(objectNodeId != null && predicateNodeId != null) {
+            createEdge(graphEdges, objectNodeId, predicateNodeId, eventGraphId);
+        }
+        
+        if (subjectNodeId == null) {
+            logger.error("A subject is missing in one of the relations for EventGraph: " + eventGraphId);
+            
+        }
+        if (objectNodeId == null) {
+            logger.error("An object is missing in one of the relations for EventGraph: " + eventGraphId); 
+        }
```

Comment:
if either of those cases is happening, should it really just ignore this and return the predicate node id?